tests/other/test_split_module.post.js

```diff
@@ -1,12 +1,25 @@
 function saveProfileData() {
-  // Avoid offset 0 so that we don't upset the stack cookie monster.
-  var offset = 1024;
   var __write_profile = Module['asm']['__write_profile'];
   if (__write_profile) {
-    var len = Module['asm']['__write_profile'](offset, -1);
+    var capacity = 1;
+    var offset;
+    var len;
+    while (true) {
+      offset = _malloc(capacity);
+      len = __write_profile(offset, capacity);
```

Comment:
Won't this end up repeatedly re-writing the profile?     Seems a little over engineered for a test case.

Also wouldn't it make sense to use the sprintf technique of returns the amount of bytes that would have been written?

Reply:
This is already doing something something similar to `sprintf` where it returns the necessary length. I suppose I could use that to be smarter than just increasing the capacity by 2x on each iteration. It only writes the profile once on the final iteration where there is sufficient space.